DXMainClient/DXGUI/Multiplayer/GameLobby/CnCNetGameLobby.cs

```diff
@@ -1738,6 +1758,79 @@ private void HandleMapSharingBlockedMessage(string sender)
                 "they will be unable to participate in this match.".L10N("UI:Main:PlayerMissingMaDisabledSharing"), sender));
         }
 
+        /// <summary>
+        /// Download a map from CNCNet using a map hash ID.
+        ///
+        /// Users and testers can get map hash IDs from this URL template:
+        ///
+        /// - http://mapdb.cncnet.org/search.php?game=GAME_ID&search=MAP_NAME_SEARCH_STRING
+        ///
+        /// </summary>
+        /// <param name="parameters">
+        /// This is a string beginning with the sha1 hash map ID, and (optionally) the name to use as a local filename for the map file.
+        /// Every character after the first space will be treated as part of the map name.
+        ///
+        /// "?" characters are removed from the sha1 due to weird copy and paste behavior from the map search endpoint.
+        /// </param>
+        private void DownloadMapByIdCommand(string parameters)
+        {
+            string sha1;
+            string mapName;
+            string message;
+
+            // Make sure no spaces at the beginning or end of the string will mess up arg parsing.
+            parameters = parameters.Trim();
+            // Check if the parameter's contain spaces.
+            // The presence of spaces indicates a user-specified map name.
+            int firstSpaceIndex = parameters.IndexOf(' ');
+
+            if (firstSpaceIndex == -1)
+            {
+                // The user did not supply a map name.
+                sha1 = parameters;
+                mapName = "user_chat_command_download".L10N("UI:Main:DownloadCommandDefaultMapName");
```

Comment:
Not sure if this one needs to be localized, also starting from this you started writing `DownloadCommand...` instead of `DownloadMapCommand`.

Also maybe it would be better to use `UI:Command:...` instead? or `UI:Commands:...`. Though if other commands are not following the same style it may be better to do this as a separately if there is a lot to change.

Reply:
Lol damn, I was worried I may have switched conventions. I should have double checked. 

My thought behind localizing that string is in case e.g. a Chinese user was looking at their files and wondering where they came from. I will remove it though.

Reply:
I'll switch mine to `UI:Command:` since there aren't already translations for them, so renaming them now is easier. Perhaps there should be a ticket to switch the other commands to `UI:Command` or `UI:ChatCommand`.